File: python/triangle/triangle.py
def equilateral(sides):
    return classify(sides) == 'equilateral'


def isosceles(sides):
    return classify(sides) in ('isosceles', 'equilateral')


def scalene(sides):
    return classify(sides) == 'scalene'


def classify(sides):
    a, b, c = sorted(sides)
    if a <= 0 or (a + b) <= c:
        # not a triangle
        return

    if a == b and a == c:
        return 'equilateral'
    elif a == b or a == c or b == c:
        return 'isosceles'
    else:
        return 'scalene'

File: dart/triangle/lib/triangle.dart
class Triangle {
  bool equilateral(num a, num b, num c) => _test([a, b, c], [1]);
  bool isosceles(num a, num b, num c) => _test([a, b, c], [1, 2]);
  bool scalene(num a, num b, num c) => _test([a, b, c], [3]);

  bool _test(List<num> sides, List<int> expected) {
    sides.sort();
    return sides[0] > 0
        && sides[0] + sides[1] > sides[2]
        && expected.contains(sides.toSet().length);
  }
}
